src/MapPane.js

```diff
@@ -0,0 +1,158 @@
+/* @flow */
+import React, { PropTypes, Component } from 'react'
+import { forEach, omit, uniqueId, indexOf } from 'lodash'
+
+import childrenType from './types/children'
+import mapType from './types/map'
+import paneType from './types/pane'
+
+const BLACKLIST = ['tile', 'shadow', 'overlay', 'map', 'marker', 'tooltip', 'popup']
+
+export default class MapPane extends Component {
+  static propTypes = {
+    name: PropTypes.string,
+    children: childrenType,
+    map: mapType,
+    zIndex: PropTypes.number,
+    className: PropTypes.string,
+    style: PropTypes.object,
+    pane: paneType,
+  };
+
+  static contextTypes = {
+    map: mapType,
+    pane: paneType,
+  };
+
+  static childContextTypes = {
+    pane: paneType,
+  };
+
+  constructor ({name}) {
+    super()
+
+    this.state = {
+      _isMounted: false,
+    }
+
+    this._name = name || `pane-${uniqueId()}`
+  }
+
+  getChildContext () {
+    return {
+      pane: this._name,
+    }
+  }
+
+  componentWillMount () {
+    const map = this.context.map || this.props.map
+
+    if (this._name && map && map.createPane) {
+      const existing = this.getPane()
+
+      if (!existing) {
+        map.createPane(this._name, this.getParentPane())
+      }
+    }
+  }
+
+  componentWillReceiveProps (props) {
+    this.setStyle(props)
+  }
+
+  setStyle ({ style, zIndex } = this.props) {
+    const pane = this.getPane()
+
+    if (pane) {
+      pane.style.zIndex = zIndex || 'initial'
+
+      if (style) {
+        forEach(style, (value, key) => {
+          pane.style[key] = value
+        })
+      }
+    }
+  }
+
+  componentDidMount () {
+    this.setState({
+      _isMounted: true,
+    })
+
+    this.setStyle()
+  }
+
+  componentWillUnmount () {
+    const { name } = this.props
+
+    if (name) {
+      const _name = (name + '').replace(/-*pane/gi, '')
+
+      if (indexOf(BLACKLIST, _name) >= 0) {
+        // Don't remove panes created by leaflet
+        return
+      }
+    }
+
+    // Remove the created pane
+    const pane = this.getPane()
+    pane && pane.remove &&
+    pane.remove()
+
+    const map = this.context.map || this.props.map
+
+    if (this._name && map && map._panes) {
+      map._panes = omit(map._panes, this._name)
+      map._paneRenderers = omit(map._paneRenderers, this._name)
+    }
+  }
+
+  getParentPane () {
```

Comment:
I've added `props.pane` to allow overriding `context.pane`. This is usefull when rendering Popup components that might get an incoming 'pane' as context but you want them to render inside another or leaflet's default poup pane.

Leaflet allows nested panes. `getParentPane` passes a parent pane node to `createPane` if that is the case.